### ISSUE TITLE
Two false negatives about the testing case in repo

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#2034]
Hi, I found two false negatives in this [file](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugsTestCases/src/java/bugPatterns/RE_POSSIBLE_UNINTENDED_PATTERN.java). In line 50 and 57, SpotBugs should have reported two warnings about the rule [IIO_INEFFICIENT_INDEX_OF](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#iio-inefficient-use-of-string-indexof-string-iio-inefficient-index-of), but it doesn't. Hence, I think they are false negatives. Thanks.

```java
@NoWarning("RE_POSSIBLE_UNINTENDED_PATTERN")
@ExpectWarning("IIO_INEFFICIENT_INDEX_OF")
int notBug2(String any) {
    return any.indexOf(".");  // no warning
}
@NoWarning("RE_POSSIBLE_UNINTENDED_PATTERN")
@ExpectWarning("IIO_INEFFICIENT_INDEX_OF")
int notBug22(String any) {
    return any.indexOf("|");  // no warning
}
```
